### ISSUE TITLE
Audio/Waveform out of sync in chrome

### ISSUE THREAD

**Issue by peet-j** [bbc/peaks.js#239]
Seems to be just a problem in chrome but on files of 60min plus, we're seeing a divergence of audio and the waveform by a couple of seconds.

Viewing the same files in Safari and Firefox both look fine so not sure what is happening with chrome.

Is this a known issue?

**Comment by peet-j** [comment]
Closing this. The issue turned out to be the encoding of the mp3 I was using. Chrome and Safari were determining the audio length differently.

**Comment by chrisn** [comment]
Thanks for updating us on this, James. Can you share any details about what you learned?

**Comment by peet-j** [comment]
The giveaway was noticing that the total audio length was displaying differently in the audio player in Chrome and Safari:

<img width="234" alt="screen shot 2018-09-26 at 13 43 51" src="https://user-images.githubusercontent.com/33906462/46085948-15881200-c19f-11e8-83e0-99d4b1c5dba6.png">
<img width="262" alt="screen shot 2018-09-26 at 13 43 42" src="https://user-images.githubusercontent.com/33906462/46085950-15881200-c19f-11e8-94e8-2ac83faa8acc.png">

This drew me to this stackoverflow post: https://stackoverflow.com/questions/20711488/html5-audio-tag-showing-wrong-duration-of-mp3-in-chrome

So I ran the mp3 through ffmpeg:
`ffmpeg -i c1d0c29f-49fd-4b9c-95e2-5fda590755a2.mp3 -acodec copy c1d0c29f-49fd-4b9c-95e2-5fda590755a2-updated.mp3` 
then replaced the old audio file with the updated one and low and behold, both chrome and safari now displayed the same audio length and the waveform played in sync.

Still looking into exactly what ffmpeg changed, but happy to know that the issue wasn't in the peak library!

Thanks for all your work @chrisn

**Comment by BuddyLReno** [comment]
@peet-j Thanks for sharing your solution. We've run into this exact issue.

**Comment by chrisn** [comment]
@BuddyLReno It's still on my to-do list to figure out exactly what's happening with this.

**Comment by BuddyLReno** [comment]
@chrisn Yeah, we're really digging in trying to figure out what FFMPEG is doing. Our files are encoded with the Fraunhofer Pro codec but I'm not sure what is exactly happening when ffmpeg recopies the stream. I don't know if it's using a different codec (i'm sure it is) or what, but it looks like all `-acodec copy` is doing is straight passing through the audio and writing the output to the new file.

I'm guessing it's when it parses the audio track through whatever codec or function it's using to read the audio.

**Comment by BuddyLReno** [comment]
@chrisn @peet-j what's interesting is that when we run the same command for ffmpeg on our mp3s, the audio is fixed in chrome but horribly out of sync in safari.  Could use some help. I'm not exactly sure where to look anymore. We've tried to generate the waveform after scrubbing the mp3 file but that didn't help either.

**Comment by BuddyLReno** [comment]
So interestingly, at least with our files, if we reencode the files with `-c:a mp3` and wrap them in wav `audio.wav`, then both safari and chrome will sync successfully.  The only conclusion I can come to at this point is that both browsers implement mp3 differently in someway.

**Comment by chrisn** [comment]
Re-opening this issue. I agree with your conclusion. If we can figure out how they differ, or at least provide some specific examples to demonstrate the differences, I'll be happy to file bug reports with the browsers. One difference I have found is related to https://github.com/bbc/audiowaveform/issues/49. Chrome skips information frames, but Firefox does not. (I don't have access to a Mac to test Safari at the moment.)

**Comment by BuddyLReno** [comment]
@chrisn how can I help you test Safari and other information that I can dig up?  I'm really pushing the boundaries of my knowledge on this topic so I'll need a little direction on how to properly test it.

I'm on a mac with Mojave so I can provide some info from the other side of the fence.
